player/scripting.c

```diff
@@ -292,32 +292,57 @@ bool mp_load_scripts(struct MPContext *mpctx)
 
 #if HAVE_CPLUGINS
 
+#if _WIN32
+#include <windows.h>
+#else
 #include <dlfcn.h>
+#endif
 
 #define MPV_DLOPEN_FN "mpv_open_cplugin"
 typedef int (*mpv_open_cplugin)(mpv_handle *handle);
 
 static int load_cplugin(struct mp_script_args *args)
 {
+    #if _WIN32
+    HMODULE lib = LoadLibraryA(args->filename);
+    #else
     void *lib = dlopen(args->filename, RTLD_NOW | RTLD_LOCAL);
+    #endif
+
     if (!lib)
         goto error;
     // Note: once loaded, we never unload, as unloading the libraries linked to
     //       the plugin can cause random serious problems.
+    #if _WIN32
+    mpv_open_cplugin sym = (mpv_open_cplugin)GetProcAddress(lib, MPV_DLOPEN_FN);
+    #else
     mpv_open_cplugin sym = (mpv_open_cplugin)dlsym(lib, MPV_DLOPEN_FN);
+    #endif
+
     if (!sym)
         goto error;
     return sym(args->client) ? -1 : 0;
 error: ;
+    #if _WIN32
+    DWORD err = GetLastError();
+    if (err)
+        MP_ERR(args, "C plugin error: '%lu'\n", err);
+    #else
     char *err = dlerror();
     if (err)
         MP_ERR(args, "C plugin error: '%s'\n", err);
+    #endif
     return -1;
```

Comment:
So I think it'd be better to either 1. duplicate the whole loader in a windows version, or 2. provide abstraction wrappers for dlopen etc. for windows. I picture 2. as providing a POSIX-like API on win32, probably messed into osdep/io.c.

Reply:
For 2. do you mean something an abstraction like mpv_{dlopen,dlsym,dlerror}?
Or defining dlopen, dlysym and dlerror on win32 only?

Anyway, I recently noticed that DLLs with non ascii characters fails to load.
I will switch from LoadLibraryA to LoadLibraryW

Reply:
Basically, the latter, but see io.h and io.c.

Reply:
Okay, I've uploaded those changes.
For the time being, I won't touch anything related to the build system even though cplugins autodetection could be enabled, provided we disable `-rdynamic` compiler option on windows.

Furthermore, I don't know if already present (dynamic build aside), we should introduce an option to build mpv.exe as a stub for mpv-1.dll.